### ISSUE TITLE
[pkgconfig] Vcpkg integrate does not work at all.

### ISSUE THREAD

**Issue by acidtonic** [microsoft/vcpkg#20689]
Installed gtkmm for the x64-windows triplet.

Ran vcpkg integrate.

I am pointing my cmake toolchain to the file it mentioned.... specifically changed my pkg_check_modules() to find_package() like the docs say and it doesn't work.

passing -DCMAKE_TOOLCHAIN_FILE and -DVCPKG_TARGET_TRIPLET="x64-windows".

Still doesn't find gtkmm-4.0 that it built and has the .pc file for. Am I supposed to override pkg-config too or is that supposed to work unmodified with that toolchain file?

**Comment by dg0yt** [comment]
I think you still have to provide some `pkg-config` executable, e.g. `pkgconf` from that vcpkg port, via cmake or environment variables.

**Comment by thehans** [comment]
I was also having issues getting pkg-config to work in my own project, even though it is used to build various vcpkg-provided dependencies which my project depends on.

I ended up installing pkgconf, which is technically a different project than `pkg-config`, but still meant to be compatible. 
The only issue is that the exe filename provided is `pkgconf.exe` instead of `pkg-config.exe`, and since cmake's builtin `FindPkgConfig` only seems to look for `pkg-config.exe`, it fails to find it.

```
vcpkg install pkgconf
```

As a workaround I provide both exe names to find and set `PKG_CONFIG_EXECUTABLE` in my project's `CMakeLists.txt`, before calling `find_package` (i'm not sure if it even matters to call `find_package` at that point, but at least it doesn't seem to hurt):
```
  find_program(PKG_CONFIG_EXECUTABLE NAMES pkgconf.exe pkg-config.exe)
  find_package(PkgConfig REQUIRED)
```

**Comment by JackBoosY** [comment]
@thehans Yes, for pkgconfig, vcpkg still need to handle something and provides the usage in output and toolchain.

**Comment by Neumann-A** [comment]
@thehans Then your `cmake` version is lower than 3.22. 3.22 added `pkgconf` as a search name.

**Comment by thehans** [comment]
@Neumann-A I'm just using whatever cmake is bundled with VS2019 and VS2022 respectively.

**Comment by JackBoosY** [comment]
Will be fixed partly by #21456.

**Comment by BillyONeal** [comment]
I believe this is a "won't fix forever" situation: MSBuild has no equivalent of reading of a .pc file or CMake Config files, so if a port needs / expects those to apply to function, msbuild customers can't use that port.

MSBuild integration works fine with all libraries that follow normal "your headers go in include, your libs go in lib, and your binaries go in bin" conventions.

Without MSBuild supporting the equivalent of pkgconfig or CMake Configs inside itself there's little we can do.
